### PR TITLE
implemented basic Elm template

### LINKED ISSUE #1: Elm Template for Embark
### Outline

The goal of this task is to create a ready to use Elm template for [Embark](https://embark.status.im/docs/). The template should have:

* The necessary packages in package.json for Elm
* A *simple* sample code using Elm
* Additions/Modifications in the (embark's) webpack to support Elm

See an example of a existing template with a custom webpack config such as the typescript template available [here](https://github.com/embark-framework/embark-typescript-template). 

This task is meant for the embark version currently in the develop branch, to install:

```
git clone git@github.com:embark-framework/embark.git
cd embark
npm install
npm link
```

starting point:
```
embark new MyApp
cd MyApp
embark eject-webpack
```

### Acceptance Criteria
* Running this template with `embark run` should result in a usable dapp using Elm as minimal starting point to start using a dapp using Elm and embark.
* It has to use embark and its pipeline (configured in `embark.json` (not elm cmd line tools) and using the embark's webpack config file (e.g found [here](https://github.com/embark-framework/embark-ember-template/blob/master/webpack.config.js)).

### PR DESCRIPTION
**Pull request by evgeniuz**
Accidentally force pushed master branch and closed previous request (https://github.com/embark-framework/embark-elm-template/pull/2).

Fixes #1.

### CODE REVIEW

**Comment by evgeniuz** [comment]
It's simple template that shows "Loading..." until embark is connected, then switches to "Ready" or showing error (to demonstrate interaction with JS/Embark through ports).

**Comment by evgeniuz** [comment]
As for RPC/WS: I've based template on running `embark new MyApp` and that was configuration there, so I don't think it should be changed here, as it's not related to Elm migration, but I can fix it if needed.

**Review by jrainville — COMMENTED** [review]
Looks good.
I would go for WS either way. We'll update our own template because it should use WS by default now.
Also, I saw that there's a directory called `elm-stuff/`, can we rename that to something more significant or it's ELM that generates that?

**Comment by evgeniuz** [comment]
This is Elm's equivalent of `node_modules` and that name is by default :)

Will change to use WS and update PR.

**Review by jrainville — APPROVED** [review]
Good work!

File: test/contract_spec.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    let result = await SimpleStorage.methods.get().call();
+    assert.strictEqual(parseInt(result, 10), 150);
+  });
+
+  it("should have account with balance", async function() {
+    let balance = await web3.eth.getBalance(accounts[0]);
+    assert.ok(parseInt(balance, 10) > 0);
+  });
+}
+*/
```
Comment (0x-r4bbit) [inline test/contract_spec.js:43]: Is this commented on purpose?

**Review by 0x-r4bbit — CHANGES_REQUESTED** [review]
Reviewed this and it looks good to me at a first glance. I'm no expert in Elm so I could only judge by common sense.

I noticed that all tests where commented, but it doesn't seem like the right thing to do. Is that a left over that should be changed?

File: test/contract_spec.js (hunk 1 above)
Comment (jrainville) [reply test/contract_spec.js:43]: Yeah, we comment the tests out in the templates so that users have a starting point when writing a new test.  Since there is no contract, that's why we comment it.

Comment (evgeniuz) [reply test/contract_spec.js:43]: I've implemented it by starting from `embark new MyApp`, like was requested in issue. These test apparently were already commented there for some reason. What I did is: created and ejected new Embark app and rewrote JavaScript frontend in Elm. All other files are left exactly as is.

This file contains tests for Solidity contract, not for JavaScript interface (or Elm in this case), so it's not relevant to this implementation.

TL;DR: It was like that, I didn't touch that :)

Comment (0x-r4bbit) [reply test/contract_spec.js:43]: Thanks for clarification you two!

Ignore my comments. Dismissing my review.

**Comment by StatusSceptre** [comment]
Any update here, are we good to payout this bounty?  @jrainville @PascalPrecht

**Comment by jrainville** [comment]
All good on my part. I would merge it myself, but I don't have permission for it :sweat_smile:
You can pay anyway and wait for @iurimatias to merge.
